src/Mvc/Mvc.Razor.RuntimeCompilation/src/ChecksumValidator.cs

```diff
@@ -100,6 +101,36 @@ public static bool IsItemValid(RazorProjectFileSystem fileSystem, RazorCompiledI
         return true;
     }
 
+    private static (byte[] checksum, string algorithm) ComputeChecksum(RazorProjectItem projectItem, string checksumAlgorithm)
+    {
+        if (projectItem == null)
+        {
+            throw new ArgumentNullException(nameof(projectItem));
+        }
+
+        Func<HashAlgorithm> createHashAlgorithm;
+        string algorithmName;
+
+        //only SHA1 and SHA256 are supported.  Default to SHA1
+        switch (checksumAlgorithm)
+        {
+            case nameof(SHA256):
+                createHashAlgorithm = SHA256.Create;
+                algorithmName = nameof(SHA256);
+                break;
+            default:
+                createHashAlgorithm = SHA1.Create;
+                algorithmName = nameof(SHA1);
+                break;
+        }
+
+        using (var stream = projectItem.Read())
+        using (var hashAlgorithm = createHashAlgorithm())
```

Comment:
Can the one-shot hashes be used instead? E.g. [`SHA256.HashData`](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.sha256.hashdata).
Avoids the allocation of the instance.

Reply:
Good idea.  I committed it.  This is my first PR, I am not sure if I am the one supposed to resolve the conversion or not

Reply:
The point is addressed so you can resolve it -- or leave it as is, just as you like.